tests/Microsoft.Identity.Test.Unit/PublicApiTests/MtlsPopTests.cs

```diff
@@ -0,0 +1,522 @@
+
+                using (var httpManager = new MockHttpManager())
+                {
+                    var app = ConfidentialClientApplicationBuilder.Create(TestConstants.ClientId)
+                        .WithCertificate(s_testCertificate)
+                        .WithAuthority($"{authorityUrl}/common")
+                        .WithAzureRegion(ConfidentialClientApplication.AttemptRegionDiscovery)
+                        .WithExperimentalFeatures()
+                        .WithHttpManager(httpManager)
+                        .BuildConcrete();
+
+                    // Expect an exception due to using /common with MTLS PoP
+                    MsalClientException ex = await Assert.ThrowsExceptionAsync<MsalClientException>(async () =>
+                        await app.AcquireTokenForClient(TestConstants.s_scope)
+                            .WithMtlsProofOfPossession()
+                            .ExecuteAsync()
+                            .ConfigureAwait(false))
+                        .ConfigureAwait(false);
+
+                    Assert.AreEqual(MsalError.AuthorityHostMismatch, ex.ErrorCode);
+                    Assert.AreEqual(MsalErrorMessage.MtlsCommonAuthorityNotAllowedMessage, ex.Message);
+                }
+            }
+        }
+
+        [TestMethod]
+        public async Task MtlsPop_ValidateExpectedUrlAsync()
+        {
+            string authorityUrl = "https://login.microsoftonline.com/123456-1234-2345-1234561234";
+
+            using (var envContext = new EnvVariableContext())
+            {
+                // Arrange
+                Environment.SetEnvironmentVariable("REGION_NAME", EastUsRegion);
+
+                using (var harness = new MockHttpAndServiceBundle())
+                {
+                    var tokenHttpCallHandler = new MockHttpMessageHandler()
+                    {
+                        ExpectedUrl = $"https://eastus.mtlsauth.microsoft.com/123456-1234-2345-1234561234/oauth2/v2.0/token",
+                        ExpectedMethod = HttpMethod.Post,
+                        ResponseMessage = CreateResponse(tokenType : "mtls_pop"),
+                        ExpectedPostData = new Dictionary<string, string>
+                        {
+                            { OAuth2Parameter.ClientId, "d3adb33f-c0de-ed0c-c0de-deadb33fc0d3" },
+                            { OAuth2Parameter.Scope, TestConstants.s_scope.AsSingleString() },
+                            { OAuth2Parameter.GrantType, OAuth2GrantType.ClientCredentials },
+                            { "token_type", "mtls_pop" }
+                        },
+                        UnExpectedPostData = new Dictionary<string, string>
+                        {
+                            { "client_assertion_type", "urn:ietf:params:oauth:client-assertion-type:jwt-bearer" },
+                            { "client_assertion", "eyJhbGciOiJQUzI1NiIsInR5cCI6IkpXVCIsIng1dCNTMjU2IjoiSnBmTm1PM1lpR2pHQ1pWY..." }
```

Comment:
put some string instead? 
```suggestion
                            { "client_assertion", "assertion." }
```